doc-changelog/action.yml

```diff
@@ -0,0 +1,138 @@
+name: >
+  Documentation Changelog
+
+description: |
+  Adds a newsfragment to the target repository using ``towncrier``
+  containing the title and number of the pull request.
+
+
+inputs:
+
+  # Required inputs
+
+  path-to-fragment:
+    description: >
+      Path to where the towncrier fragment is stored
+    required: true
+    type: string
+
+  pr-branch:
+    description: >
+      Name of the pull request
+    required: true
+    type: string
+
+  pr-title:
+    description: >
+      Title of the pull request
+    required: true
+    type: string
+
+  pr-number:
+    description: >
+      The pull request number
+    required: true
+    type: string
+
+  token:
+    description: >
+      GitHub token for doing a git commit & push
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: "Install Git and clone project"
+      uses: actions/checkout@v4
+      with:
+        ref: ${{ inputs.pr-branch }}
+
+    - name: "Set up Python ${{ inputs.python-version }}"
+      uses: ansys/actions/_setup-python@main
+      with:
+        python-version: ${{ inputs.python-version }}
+        use-cache: ${{ inputs.use-python-cache }}
+
+    - name: "Install towncrier"
+      shell: bash
+      run: |
+        python -m pip install --upgrade pip towncrier
+
+    - name: "Get pull request type"
+      shell: bash
+      run: |
+        # Check inputs.pr-title is surrounded by double quotes
+        title='${{ inputs.pr-title }}'
+        if [[ "$title" =~ ^\".*\"$ ]]; then
+          echo "PR title is wrapped in double quotes"
+        else
+          echo "PR title is not wrapped in double quotes"
+          title="${{ inputs.pr-title }}"
+        fi
+
+        # Save title into env variable
+        echo "PR-TITLE=$title" >> $GITHUB_ENV
+
+        # Check if pr-title string starts with Add, Remove, Change, Fix, or Bump
```

Comment:
For this checks, we may consider using https://github.com/ansys/actions/blob/release/4.1/commit-style/action.yml. This way, we enforce pull-requests to follow conventional commits.

Reply:
That would be great! I didn't know that existed